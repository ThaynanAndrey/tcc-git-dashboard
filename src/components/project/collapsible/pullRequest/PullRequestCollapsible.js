import React from 'react';
import Collapsible from '../../../layout/Collapsible';
import CollapsibleCard from '../../../layout/CollapsibleCard';
import PullRequestCollapsibleHeader from './PullRequestCollapsibleHeader';
import PullRequestCollapsibleBody from './PullRequestCollapsibleBody';

/**
 * Stateless component with the function of presenting a Collapsible to Pull Requests.
 * 
 * @author Thaynan Nunes
 */
const PullRequestCollapsible = ({ idProject, redirectPage, openPullRequest }) => (
    <Collapsible>
        <CollapsibleCard isActive={true}>
            
            <PullRequestCollapsibleHeader
                idProject={idProject}
                redirectPage={redirectPage}
            />

            <PullRequestCollapsibleBody
                idProject={idProject}
                openPullRequest={openPullRequest}
            />

        </CollapsibleCard>
    </Collapsible>
);

export default PullRequestCollapsible;