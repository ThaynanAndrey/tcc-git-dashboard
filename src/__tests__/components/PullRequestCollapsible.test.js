import React from 'react';
import { shallow, configure } from "enzyme";
import toJson from 'enzyme-to-json';

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import PullRequestCollapsible from '../../components/project/collapsible/pullRequest/PullRequestCollapsible';

describe('Test RepositoriesNoProject', () => {
  it('renders correctly a component to show PullRequestCollapsible', () => {
    const [redirectPage, openPullRequest] = new Array(2).fill(jest.fn());
    const idProject = '123';

    const tree = shallow(
      <PullRequestCollapsible 
        idProject={idProject}
        redirectPage={redirectPage}
        openPullRequest={openPullRequest}
      />
    );

    expect(toJson(tree)).toMatchSnapshot();
  });
})