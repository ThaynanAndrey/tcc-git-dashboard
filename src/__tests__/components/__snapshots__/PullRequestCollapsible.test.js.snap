// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test RepositoriesNoProject renders correctly a component to show PullRequestCollapsible 1`] = `
<Collapsible>
  <CollapsibleCard
    isActive={true}
  >
    <PullRequestCollapsibleHeader
      idProject="123"
      redirectPage={[MockFunction]}
    />
    <PullRequestCollapsibleBody
      idProject="123"
      openPullRequest={[MockFunction]}
    />
  </CollapsibleCard>
</Collapsible>
`;
